cuda_core/tests/test_memory.py

```diff
@@ -437,3 +441,11 @@ def test_mempool_attributes_ownership(mempool_device):
     with pytest.raises(RuntimeError, match="DeviceMemoryResource is expired"):
         _ = attributes.used_mem_high
     mr._mempool_handle = old_handle
+
+
+# Ensure that memory views dellocate their reference to dlpack/cupy tensors
+@pytest.mark.skipif(cupy is None, reason="cupy is not installed")
+def test_strided_memory_view_leak(benchmark):
+    for idx in range(1000):
+        arr = cupy.zeros((1024, 1024, 1024), dtype=cupy.uint8)
+        StridedMemoryView(arr, stream_ptr=-1)
```

Comment:
Instead of doing 1000 iterations, we might be able to do something much smaller and introspect the cupy memory pool to ensure memory is being freed as expected: https://docs.cupy.dev/en/stable/reference/generated/cupy.cuda.MemoryPool.html#cupy.cuda.MemoryPool

Or if we're feeling ambitious we could use a temporary custom allocator for CuPy that we could use to track the allocations and deallocations: https://docs.cupy.dev/en/stable/reference/generated/cupy.cuda.using_allocator.html